lib/Button.js

```diff
@@ -147,99 +168,107 @@ export default class Button extends Component {
         const textStyle = (() => {
             if (disabled || !(overrides && overrides.textColor)) {
                 return textStyleMap[shape][theme][type];
+            } else {
```

Comment:
@codenamev if you could keep it inline with what @Ehesp mentioned re consistent returns then i'll merge this PR, cheers!